File: twoSum.js
{
    const twoSum = function(nums, target) {
        for(let i = 0; i < nums.length; i++) {
            for(let j = i + 1;j < nums.length; j++) {
                if(nums[i] + nums[j] === target) {
                    return [i,j]
                }
            }
        }

        return []
    };

    // const twoSum = function(nums, target) {
    //     console.log(nums)

    //     for(let i = 0;i < nums.length; i++) {
    //         console.log(nums[i])
    //     }
    // };

    console.log(twoSum([2,5,5,11], 10))
    // console.log(twoSum([3,2,4],6))
    // console.log(twoSum([3,3],6))
}

File: two-sum.rs
pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let length = nums.len();

    for first_num in 1..length {
        for second_num in first_num..length {
            let left_side = second_num - first_num;
            let right_side = second_num;
            if nums[left_side] + nums[right_side] == target {
                return vec![left_side as i32, right_side as i32];
            }
        }
    }
    vec![]
}
